File: backend/apps/orcamentos/services/ncm_investimento.py
"""NCM exibido na tabela de investimento (perfil solução completa)."""
from __future__ import annotations

from apps.orcamentos.constants import NCM_INVESTIMENTO_PAINEL_PADRAO


def normalizar_ncm_investimento(valor: str | None, *, padrao: str = NCM_INVESTIMENTO_PAINEL_PADRAO) -> str:
    digitos = "".join(ch for ch in (valor or "") if ch.isdigit())
    return digitos or padrao


def ncm_investimento_orcamento(orcamento) -> str:
    return normalizar_ncm_investimento(getattr(orcamento, "ncm_investimento", None))

File: frontend/src/modules/orcamentos/utils/ncmInvestimento.ts
/** NCM padrão na oferta — painéis elétricos (perfil solução completa). */
export const NCM_INVESTIMENTO_PAINEL_PADRAO = '85371090'

export function normalizarNcmInvestimento(valor: string | null | undefined): string {
  const digitos = (valor ?? '').replace(/\D/g, '')
  return digitos || NCM_INVESTIMENTO_PAINEL_PADRAO
}

export function formatarNcmInvestimentoInput(valor: string): string {
  return valor.replace(/\D/g, '').slice(0, 8)
}
